src/lib/fetchAPI.js

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import Cookies from "js-cookie";
+import toast from "react-hot-toast";
+import { useState } from "react";
+
+export const useRequestLogin = () => {
+  const router = useRouter();
+  const [loading, setLoading] = useState(false);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  async function handleSubmitLogin(e) {
+    try {
+      setLoading(true);
+      //prevent refresh on submissions
+      e.preventDefault();
+      //fetching the data from the API
+      const res = await fetch("https://eventmakers-api.fly.dev/auth/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email,
+          password,
+        }),
+      });
```

Comment:
Hi, please only put this part into a `requestLogin()` function in /lib/fetchAPI.

The rest of the hook can be put inside `/LoginForm.jsx`